### PR TITLE
doc: Clarify the existence of client_os

### PR DESCRIPTION
**Pull request by untitaker**
See also https://www.notion.so/sentry/Conflicting-OS-contexts-6b4b157034c54c02a97ea8523bee265d#6aa1e8cc97f4444bb33d180ecf7c3adf

### CODE REVIEW

**Comment by bruno-garcia** [comment]
Adding @mattjohnsonpint just for his info but he's OOO for a week

**Review by jjbayer — COMMENTED** [review]
Thanks for documenting this! While it's good for clarifying the current situation, in the long run I would not mention `server_os`, or even the possibility to create keys other than `os` or `client_os`.

**Comment by untitaker** [comment]
>While it's good for clarifying the current situation, in the long run I would not mention server_os, or even the possibility to create keys other than os or client_os.

I think that people do not only look at these docs to figure out what to send, but also to explain what they're seeing in the JSON that they can download from the server. SDK devs are sometimes curious how they can replicate what they see in web frontend events and generally do review how the event looks in issue details

**Review by jjbayer — APPROVED** [review]

**Review by jjbayer — APPROVED** [review]

File: src/docs/sdk/event-payloads/contexts.mdx [hunk 1]
```diff
@@ -196,11 +196,44 @@ The `type` and default key is `"device"`.
...
-User-Agent string).
+OS context under the default key `"os"` describes the operating system on which
+the crash happened/the event was created.
 
-The `type` and default key is `"os"`.
+The `type` and default key is `"os"`. However, since contexts can be set
+multiple times under different keys, there has historically been a lot of
+confusion about which OS context represents what. So here's some examples:
+
+* In events reported from a Python/ASPNET/Rails web backend, the OS context
```
Comment (bruno-garcia) [inline src/docs/sdk/event-payloads/contexts.mdx:206]:
```suggestion
* In events reported from a Python/ASP.NET/Rails web backend, the OS context
```

File: src/docs/sdk/event-payloads/contexts.mdx [hunk 2]
```diff
@@ -196,11 +196,44 @@ The `type` and default key is `"device"`.
...
+* In events reported from a JS web frontend, the SDK typically reports no OS
+  context.
+
+  The server however knows by looking at the platform (`"javascript"`) that the
+  incoming User-Agent can only come from the crashing device, and creates the
+  User-Agent based OS context under the default key `"os"`.
+
+To summarize:
+
+* `"os"` key for the crashing device's OS.
```
Comment (bruno-garcia) [inline src/docs/sdk/event-payloads/contexts.mdx:223]:
```suggestion
* `"os"` key for the device generating the event.
```

File: src/docs/sdk/event-payloads/contexts.mdx [hunk 3]
```diff
@@ -196,11 +196,44 @@ The `type` and default key is `"device"`.
...
+* `"client_os"` key for an adjacent client device's OS (that is _not_ the
+  crashing device) if it's important. The Sentry server sets this as part of
+  User-Agent parsing, but SDKs can set this too.
```
Comment (bruno-garcia) [inline src/docs/sdk/event-payloads/contexts.mdx:226]:
```suggestion
* `"client_os"` key for an adjacent client device's OS (that is _not_ the
  device creating the Sentry event instance) if it's important. The Sentry server sets this as part of
  User-Agent parsing, but SDKs can set this directly too.
```

**Review by bruno-garcia — APPROVED** [review]
Thanks Markus. Left some suggestions: Prefer "the device creating the event" as opposed to "crashing device" since on the case of server, we can be creating a transaction or message or 'error' event so the word 'crashing' here is a bit off

**Pull request merged**